lib/driver.js

```diff
@@ -162,6 +162,9 @@ class XCUITestDriver extends BaseDriver {
     }
 
     let { device, udid, realDevice } = await this.determineDevice();
+    if (!realDevice && this.opts.scaleFactor !== undefined) {
```

Comment:
Can this new bit be moved below the logging and setting of the device information? Separating the getting of device and using what was retrieved makes telling what is going on a bit more difficult.